### PR TITLE
[CFE-416]: Feature: contract detail

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/services/tx/index.ts [hunk 1]
```diff
@@ -119,7 +120,8 @@ export const useTxsByAddress = (
   isSigner: Option<boolean>,
   txFilters: TxFilters,
   offset: number,
-  limit: number
+  limit: number,
```
Comment (evilpeach) [inline src/lib/services/tx/index.ts:123]: limit, offset

File: src/lib/pages/contract-details/components/tables/TxsTable.tsx [hunk 2]
```diff
@@ -35,14 +39,28 @@ export const TxsTable = ({
     },
   });
 
-  const { data, isLoading, error } = useTxsByAddress(
+  const txsByAddressFull = useTxsByAddress(
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/tables/TxsTable.tsx:42]: txsByAddressFull => resApi

File: src/lib/pages/contract-details/components/tables/TxsTable.tsx [hunk 3]
```diff
@@ -35,14 +39,28 @@ export const TxsTable = ({
...
     contractAddress,
     undefined,
     undefined,
     DEFAULT_TX_FILTERS,
     offset,
-    pageSize
+    pageSize,
+    { enabled: isFullTier }
   );
+  const txsByAddressLite = useTxsByContractAddressLcd(
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/tables/TxsTable.tsx:51]: resLcd

File: src/lib/pages/contract-details/components/tables/TxsTable.tsx [hunk 4]
```diff
@@ -56,7 +74,7 @@ export const TxsTable = ({
             <EmptyState
               withBorder
               imageVariant="empty"
-              message="This contract does not have any transactions"
+              message="This contract does not have any transactions, or they are too old and have been pruned from the LCD."
```
Comment (evilpeach) [inline src/lib/pages/contract-details/components/tables/TxsTable.tsx:77]: Empty message should depends on isFullTier.

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**